android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java

```diff
@@ -977,9 +979,20 @@ public void onPackageInstalled(List<Map<String, String>> keyboardsInstalled) {
           aPreparedCloudApiParams.add(new CloudApiTypes.CloudApiParam(
             CloudApiTypes.ApiTarget.KeyboardLexicalModels, url).setType(CloudApiTypes.JSONType.Array));
 
-          CloudDownloadMgr.getInstance().executeAsDownload(
-            context, _downloadid, null, _callback,
-            aPreparedCloudApiParams.toArray(new CloudApiTypes.CloudApiParam[0]));
+          Toast errorToast = Toast.makeText(context,
+            context.getString(com.keyman.engine.R.string.update_check_unavailable),
```

Comment:
We shouldn't be using resources from Keyman Engine in Keyman app.

Also, it'd be cleaner for this to only be constructed in an error scenario.

Finally, this says 'update_check_unavailable' which gives 'Failed to access server' but surely for `DownloadManagerDisabledException` we can do better than that and actually give a pointer to what is going wrong ("Failed to initialize download: DownloadManager app appears to be disabled" or something like that?)

Reply:
Interestingly, one existing case of that prior use (`update_check_unavailable`, via `ResourcesUpdateTool`) will _intercept_ this suggested (more distinct) error in the startup-crashing cases.  It may take notable extra work to have a distinct error reported for those cases.

Reply:
Related code blocks:

https://github.com/keymanapp/keyman/blob/aa6151032583399303619b20a7e0ea9e926fd818/android/KMEA/app/src/main/java/com/keyman/engine/logic/ResourcesUpdateTool.java#L102

https://github.com/keymanapp/keyman/blob/aa6151032583399303619b20a7e0ea9e926fd818/android/KMEA/app/src/main/java/com/keyman/engine/logic/ResourcesUpdateTool.java#L119-L130

Sadly, the `ResourcesUpdateTool` and the methods it invokes expect `Runnable` instances for handling failures - a structure that doesn't permit parameterization.  During this work, it was clear that this "failure" handler should also handle `DownloadManager` disablement - that's the sort of thing a failure handler is there to do.  But... it's really hard to make this properly picky about which message it shows.

Reply:
> We shouldn't be using resources from Keyman Engine in Keyman app.

The refactor suggestion in a different review comment of yours - the one to make a helper to centralize the error-case handling - helped resolve this aspect swimmingly.  It'll also allow for the custom message to display in-app for cases _other_ than resource-update checks.